build-functions.php

```diff
@@ -29,7 +29,7 @@ function kebabCaseToUcWords($kebab)
  */
 function toKebabCase($string)
 {
-    return implode('-', explode(' ', strtolower($string)));
+    return preg_replace('/[ _\\-\\+\\/\\\\]+/', '-', strtolower($string));
```

Comment:
Didn't understand the second part, after "and then collapse ...".

But note that the example on the site needs to be de-escaped. [See now](http://www.phpliveregex.com/p/kCB).